include/boost/multiprecision/jacobi.hpp

```diff
@@ -0,0 +1,100 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2018-2020 Mikhail Komarov <nemo@nil.foundation>
+//
+// Distributed under the Boost Software License, Version 1.0
+// See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt
+//---------------------------------------------------------------------------//
+
+#ifndef BOOST_MULTIPRECISION_JACOBI_HPP
+#define BOOST_MULTIPRECISION_JACOBI_HPP
+
+#include <boost/multiprecision/detail/default_ops.hpp>
+
+namespace boost {
+namespace multiprecision {
+
+template <typename Backend>
+inline int eval_jacobi(const Backend& a, const Backend& n)
+{
+   using default_ops::eval_get_sign;
+   using default_ops::eval_gt;
+   using default_ops::eval_integer_modulus;
+   using default_ops::eval_is_zero;
+   using default_ops::eval_lsb;
+   using default_ops::eval_lt;
+   using default_ops::eval_modulus;
+   using default_ops::eval_right_shift;
+
+   if (eval_get_sign(a) < 0)
+   {
+      BOOST_THROW_EXCEPTION(std::invalid_argument("jacobi: first argument must be non-negative"));
+   }
+   if (!eval_integer_modulus(n, 2) || eval_lt(n, 2))
+   {
+      BOOST_THROW_EXCEPTION(std::invalid_argument("jacobi: second argument must be odd and > 1"));
+   }
+
+   Backend   x = a, y = n;
+   limb_type J = 1;
```

Comment:
limb_type is an unsigned type, and is not guaranteed to interoperate with anything except cpp_int anyway, try `typename mpl::front<typename Backend::signed_types>::type`.  (and yes I realise that's a mouthful!)

Reply:
Well, there is no need in storing some data in the actual `limb_type` so I've just replaced it with `int`. Should work fine.

By the way, thank you for the suggestion on backend-independent way of getting the `limb_type`. I know where I'ill use it.